pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/FunctionApiResource.java

```diff
@@ -54,7 +54,7 @@ public String clientAppId() {
                 : null;
     }
 
-    public AuthenticationDataHttps clientAuthData() {
-        return (AuthenticationDataHttps) httpRequest.getAttribute(AuthenticationFilter.AuthenticatedDataAttributeName);
+    public AuthenticationDataSource clientAuthData() {
+        return (AuthenticationDataSource) httpRequest.getAttribute(AuthenticationFilter.AuthenticatedDataAttributeName);
```

Comment:
If the return type change to the interface `AuthenticationDataSource`, it indicates the return value will be different implementations. But it seems that the implementation of the interface `AuthenticationDataSource` here will always be `AuthenticationDataHttps`.

Reply:
Use AuthenticationDataSource to allow users to customize their own auth state, if the default is to use AuthenticationDataHttps as the return value, it will not allow users to customize the auth state

Reply:
Got it, thanks.